src/breedsRepo.js

```diff
@@ -1,82 +1,170 @@
-        } else {
-            for (let key in object) {
-                if (breedsCopy[index][key] !== object[key]) {
-                    breedsCopy[index][key] = object[key];
-                    console.log(`Key '${key}' successfully updated`);
-                }
+        
+        const dao2 = this.dao;
+        const checkForId = this.checkForId; 
+        const getBreedById = this.getBreedById; 
+
+        const getProm = async function () {
+            try {                
+                if (!checkForId(id)) {
+                    return null;
+                } else {
+                    // iterates through object to create a string of keys and values in format: 'column = value column2 = value2'
+                    let arr = []; 
+                    console.log('object:' + JSON.stringify(object));
+                    for (let key in object) {
+                        arr.push(` ${key} = '${object[key]}'`);   
+                    }
+                    console.log('arr: ' + arr); 
+                    let sql = `UPDATE breeds SET ${arr.join()} WHERE breedId = ?;`
+                    await dao2.run(sql,[id]);
+                    let checkNewRecord = getBreedById(id);  
+                    return checkNewRecord; 
+                };
+            } catch (err) {
+                console.log(err); 
             }
-            this.setBreeds = breedsCopy; 
-            return breedsCopy[index]; 
-        };       
+        }
+        return getProm();      
     }
 
-    addBreed(newBreed) { 
-        const breedsCopy = this.getAllBreeds;
-        const newIndex = breedsCopy.length + 1
-        newBreed.breedId = newIndex;
-        breedsCopy.push(newBreed);
-        this.setBreeds = breedsCopy; 
-        return newBreed; 
+    addBreed(object) { 
+
+        const dao2 = this.dao;
+        const getBreedById = this.getBreedById; 
+
+        const getProm = async function () {
+            try {
+                // generates new id value. 
+                let idArray = await dao2.allId('SELECT * FROM breeds;'); 
+                // while statement returns the highest missing id value in the current series of ids
+                console.log('idArray: '+  idArray);
+                let count = 1; 
+                while (count === idArray[count-1]) {
+                    count++;
+                }
+                let newId = count; 
+                console.log('new id: ' + newId);
+
+                // insert new record row 
+                await dao2.run(
+                    `INSERT INTO breeds (breedId, name,\
+                    description) VALUES (?,?,?);`, 
+                    [newId, object.name, object.description]);
+
+                // check that new record added correctly and returns new record
+                return getBreedById(newId); 
+            } catch (err) {
+                console.log(err);
+            };
+        }
+        return getProm();   
     }
 
     deleteBreedById(id) {
```

Comment:
A future task would be to consider implementing the database using a foreign key: https://www.sqlite.org/foreignkeys.html

This effectively enforces a relationship between rows in two separate tables which makes it harder to put incorrect data in.

Reply:
Agreed. I'm aware of how to view info across seperate tables using SQL JOIN statements: 
```
SELECT * FROM cats 
JOIN breeds 
ON cats.breedId =  breeds.breedId; 
```
It would be good to incorporate this into the API in some useful way. Will revisit at a later date.